File: flutter_app/lib/models/Recipient.dart
class Recipient {
  final String id;
  final String name;
  final String phoneNumber;
  final String emailAddress;

  const Recipient({
    this.id = '',
    required this.name,
    required this.phoneNumber,
    required this.emailAddress,
  });

  factory Recipient.fromJson(Map<String, dynamic> json) {
    return Recipient(
      id: json['id'] as String? ?? '',
      name: json['name'] as String? ?? '',
      phoneNumber: json['phoneNumber'] as String? ?? '',
      emailAddress: json['email'] as String? ?? '',
    );
  }
}

File: dotnet_api/Models/Recipient.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace dotnet_api.Models;

public class Recipient
{
    [Key]
    [MaxLength(50)]
    public string Id { get; set; } = Guid.NewGuid().ToString();

    [Required]
    [MaxLength(50)]
    public string ClientId { get; set; } = string.Empty;

    [Required]
    [MaxLength(120)]
    public string Name { get; set; } = string.Empty;

    [Required]
    [MaxLength(25)]
    public string PhoneNumber { get; set; } = string.Empty;

    [MaxLength(255)]
    [EmailAddress]
    public string? Email { get; set; }

    [MaxLength(255)]
    public string? Address { get; set; }

    [MaxLength(20)]
    public string? AgeRange { get; set; }

    [MaxLength(20)]
    public string? Gender { get; set; }

    public bool IsActive { get; set; } = true;

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    [ForeignKey("ClientId")]
    public virtual Client? Client { get; set; }
}
